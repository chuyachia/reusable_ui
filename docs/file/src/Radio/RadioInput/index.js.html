<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <base data-ice="baseUrl" href="../../../../">
  <title data-ice="title">src/Radio/RadioInput/index.js | reusable-components-poc</title>
  <link type="text/css" rel="stylesheet" href="css/style.css">
  <link type="text/css" rel="stylesheet" href="css/prettify-tomorrow.css">
  <script src="script/prettify/prettify.js"></script>
  <script src="script/manual.js"></script>
</head>
<body class="layout-container" data-ice="rootContainer">

<header>
  <a href="./">Home</a>
  
  <a href="identifiers.html">Reference</a>
  <a href="source.html">Source</a>
  
  <div class="search-box">
  <span>
    <img src="./image/search.png">
    <span class="search-input-edge"></span><input class="search-input"><span class="search-input-edge"></span>
  </span>
    <ul class="search-result"></ul>
  </div>
</header>

<nav class="navigation" data-ice="nav"><div>
  <ul>
    
  <li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-useTheme">useTheme</a></span></span></li>
</ul>
</div>
</nav>

<div class="content" data-ice="content"><h1 data-ice="title">src/Radio/RadioInput/index.js</h1>
<pre class="source-code line-number raw-source-code"><code class="prettyprint linenums" data-ice="content">import styled from &quot;@emotion/styled&quot;;

import useTheme from &quot;../../useTheme&quot;;
import RadioInput from &quot;./RadioInput&quot;;

export default styled(RadioInput)`
  display: ${props =&gt;
    props.direction == &quot;horizontal&quot; ? &quot;inline-block&quot; : &quot;block&quot;};

  position: relative;
  cursor: pointer;
  font-family: ${props =&gt; useTheme(props, &quot;fontSans&quot;)};
  padding-left: 1.5em;
  &amp; input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
  }
  &amp; span {
    position: absolute;
    top: 0;
    left: 0;
    height: 1em;
    width: 1em;
    background-color: ${props =&gt; useTheme(props, &quot;silver&quot;)};

    border-radius: 50%;
  }
  &amp;:hover span {
    background-color: ${props =&gt; {
      switch (props.variant) {
        case &quot;secondary&quot;:
          return useTheme(props, &quot;secondaryLight&quot;);
        case &quot;warning&quot;:
          return useTheme(props, &quot;warningLight&quot;);
        default:
          return useTheme(props, &quot;primaryLight&quot;);
      }
    }};
  }
  &amp; input:checked ~ span {
    background-color: ${props =&gt; {
      switch (props.variant) {
        case &quot;secondary&quot;:
          return useTheme(props, &quot;secondary&quot;);
        case &quot;warning&quot;:
          return useTheme(props, &quot;warning&quot;);
        default:
          return useTheme(props, &quot;primary&quot;);
      }
    }};
  }
  span:after {
    content: &quot;&quot;;
    position: absolute;
    display: none;
  }
  &amp; input:checked ~ span:after {
    display: block;
  }
  &amp; span:after {
    top: 50%;
    left: 50%;
    width: 0.5em;
    height: 0.5em;
    border-radius: 50%;
    background: ${props =&gt; useTheme(props, &quot;silver&quot;)};
    transform: translate(-50%, -50%);
  }
`;
</code></pre>

</div>

<footer class="footer">
  Generated by <a href="https://esdoc.org">ESDoc<span data-ice="esdocVersion">(1.1.0)</span><img src="./image/esdoc-logo-mini-black.png"></a>
</footer>

<script src="script/search_index.js"></script>
<script src="script/search.js"></script>
<script src="script/pretty-print.js"></script>
<script src="script/inherited-summary.js"></script>
<script src="script/test-summary.js"></script>
<script src="script/inner-link.js"></script>
<script src="script/patch-for-local.js"></script>
</body>
</html>
