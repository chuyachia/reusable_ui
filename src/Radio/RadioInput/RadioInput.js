import React from "react";
import PropTypes from "prop-types";

const RadioInput = ({ label, value, className, name }) => (
  <label className={className}>
    {label}&nbsp;
    <input type="radio" name={name} value={value} />
    <span />
  </label>
);

RadioInput.propTypes = {
  label: PropTypes.string,
  className: PropTypes.string,
  name: PropTypes.string,
  value: PropTypes.string,
};

export default RadioInput;
