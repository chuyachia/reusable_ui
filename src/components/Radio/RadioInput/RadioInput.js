import React from "react";
import PropTypes from "prop-types";

const RadioInput = ({ label, value, className, name }) => (
  <label className={className}>
    {label}
    <input type="radio" name={name} value={value}/>
    <span />
  </label>
);

RadioInput.propTypes = {
  label: PropTypes.string,
  value: PropTypes.string,
};

export default RadioInput;
