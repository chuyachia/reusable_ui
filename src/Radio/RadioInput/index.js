import styled from "@emotion/styled";

import useTheme from "../../useTheme";
import RadioInput from "./RadioInput";

export default styled(RadioInput)`
  display: ${props =>
    props.direction == "horizontal" ? "inline-block" : "block"};

  position: relative;
  cursor: pointer;
  font-family: ${props => useTheme(props, "fontSans")};
  padding-left: 1.5em;
  & input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
  }
  & span {
    position: absolute;
    top: 0;
    left: 0;
    height: 1em;
    width: 1em;
    background-color: ${props => useTheme(props, "silver")};

    border-radius: 50%;
  }
  &:hover span {
    background-color: ${props => {
      switch (props.variant) {
        case "secondary":
          return useTheme(props, "secondaryLight");
        case "warning":
          return useTheme(props, "warningLight");
        default:
          return useTheme(props, "primaryLight");
      }
    }};
  }
  & input:checked ~ span {
    background-color: ${props => {
      switch (props.variant) {
        case "secondary":
          return useTheme(props, "secondary");
        case "warning":
          return useTheme(props, "warning");
        default:
          return useTheme(props, "primary");
      }
    }};
  }
  span:after {
    content: "";
    position: absolute;
    display: none;
  }
  & input:checked ~ span:after {
    display: block;
  }
  & span:after {
    top: 50%;
    left: 50%;
    width: 0.5em;
    height: 0.5em;
    border-radius: 50%;
    background: ${props => useTheme(props, "silver")};
    transform: translate(-50%, -50%);
  }
`;
