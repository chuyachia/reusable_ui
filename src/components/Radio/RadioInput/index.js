import styled from "@emotion/styled";

import RadioInput from "./RadioInput";

export default styled(RadioInput)`
  display: ${props =>
    props.direction == "horizontal" ? "inline-block" : "block"};

  position: relative;
  cursor: pointer;
  font-family: ${props => props.theme.fontSans};
  padding-left: 1.5em;
  & input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
  }
  & span {
    position: absolute;
    top: 0;
    left: 0;
    height: 1em;
    width: 1em;
    background-color: ${props => props.theme.silver};

    border-radius: 50%;
  }
  &:hover span {
    background-color: ${props => {
      switch (props.variant) {
        case "secondary":
          return props.theme.secondaryLight;
        case "warning":
          return props.theme.warningLight;
        default:
          return props.theme.primaryLight;
      }
    }};
  }
  & input:checked ~ span {
    background-color: ${props => {
      switch (props.variant) {
        case "secondary":
          return props.theme.secondary;
        case "warning":
          return props.theme.warning;
        default:
          return props.theme.primary;
      }
    }};
  }
  span:after {
    content: "";
    position: absolute;
    display: none;
  }
  & input:checked ~ span:after {
    display: block;
  }
  & span:after {
    top: 50%;
    left: 50%;
    width: 0.5em;
    height: 0.5em;
    border-radius: 50%;
    background: ${props => props.theme.silver};
    transform: translate(-50%, -50%);
  }
`;
