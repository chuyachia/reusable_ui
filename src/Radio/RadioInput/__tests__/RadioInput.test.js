import React from "react";

import RadioInput from "../RadioInput";

describe("RadioInput", () => {
  let wrapper;
  it("should render", () => {
    wrapper = shallow(<RadioInput />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render its children with properties", () => {
    var testclassname = "testclassname";
    var testvalue = "testvalue";
    var testname = "testname";
    wrapper = shallow(
      <RadioInput className={testclassname} value={testvalue} name={testname}>
        <span>Item</span>
      </RadioInput>
    );
    expect(wrapper.find("label").prop("className")).toBe(testclassname);
    expect(wrapper.find("input").prop("name")).toBe(testname);
    expect(wrapper.find("input").prop("value")).toBe(testvalue);
  });
});
