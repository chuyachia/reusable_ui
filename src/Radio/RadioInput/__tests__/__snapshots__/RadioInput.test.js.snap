// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioInput should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RadioInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        " ",
        <input
          type="radio"
        />,
        <span />,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      undefined,
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": undefined,
          "type": "radio",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          " ",
          <input
            type="radio"
          />,
          <span />,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        undefined,
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": undefined,
            "type": "radio",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
